yb-voyager/cmd/exportSchema.go

```diff
@@ -331,6 +336,82 @@ func applyMigrationAssessmentRecommendations() error {
 	return nil
 }
 
+// TODO: merge this function with applying sharded/colocated recommendation
+func applyMergeConstraintsTransformations() {
+	if utils.GetEnvAsBool("YB_VOYAGER_SKIP_MERGE_CONSTRAINTS_TRANSFORMATIONS", false) {
+		log.Infof("skipping applying merge constraints transformation due to env var YB_VOYAGER_SKIP_MERGE_CONSTRAINTS_TRANSFORMATIONS=true")
+		return
+	}
+
+	utils.PrintAndLog("Applying merge constraints transformation to the exported schema")
+	transformer := transform.NewTransformer()
+
+	fileName := utils.GetObjectFilePath(schemaDir, TABLE)
+	if !utils.FileOrFolderExists(fileName) {
+		log.Infof("table.sql file doesn't exists, skipping applying merge constraints transformation")
+		return
+	}
+
+	rawStmts, err := queryparser.ParseSqlFile(fileName)
+	if err != nil {
+		utils.ErrExit("failed to parse table.sql file: %v", err)
+	}
+
+	// filter SET statements(PostgreSQL) from rawStmts
+	tableStmts := lo.Filter(rawStmts.Stmts, func(stmt *pg_query.RawStmt, _ int) bool {
+		stmtType := queryparser.GetStatementType(stmt.Stmt.ProtoReflect())
+		return stmtType != queryparser.PG_QUERY_VARIABLE_SET_STMT_NODE && stmtType != queryparser.PG_QUERY_SELECTSTMT_NODE
+	})
+
+	// TODO: separate handling for Oracle and MySQL ??
+	sessionVarStmts := lo.Filter(rawStmts.Stmts, func(stmt *pg_query.RawStmt, _ int) bool {
+		stmtType := queryparser.GetStatementType(stmt.Stmt.ProtoReflect())
+		return stmtType == queryparser.PG_QUERY_VARIABLE_SET_STMT_NODE || stmtType == queryparser.PG_QUERY_SELECTSTMT_NODE
+	})
+
+	// NOTE: In case of any error during merge constraints or deparsing, log the error and stick to the original table.sql file
+
+	transformedRawStmts, err := transformer.MergeConstraints(tableStmts)
+	if err != nil {
+		log.Errorf("failed to merge constraints: %v", err)
```

Comment:
I think this function should return errors properly. It can do it's job as expected and throw errors wherever encountered. The caller can decide how to handle the error (throw the error futher up, or just log and ignore)